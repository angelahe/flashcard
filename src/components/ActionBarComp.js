import React from 'react';
import PropTypes from 'prop-types';
import addBtn from '../img/buttons/add_FFFFFF.png';
import backBtn from '../img/buttons/back_FFFFFF.png';
import logo from '../img/buttons/logo_FFFFFF.png';
import '../styles/flashcard.css';

const ActionBarComp = (props) => {
  const {
    showAdd,
    headerText,
    onAdd,
    onBack,
  } = props;

  function handleAddClicked() {
    console.log('in handleAddClicked of ActionBar');
    onAdd();
  }

  function handleBackClicked() {
    console.log('in handleBackClicked of ActionBar');
    onBack();
  }

  return (
    <div className="ActionBar ActionBarComp">
      <img className="NavImg NavBtn" src={logo} alt="logo" />
      { showAdd
        ? (
          <button type="button" className="NavBtn" onClick={handleAddClicked}>
            <img className="NavImg" src={addBtn} alt="Add" />
          </button>
        )
        : null
      }
      <span className="HeaderText">{headerText}</span>
      <button type="button" className="NavBtn" onClick={handleBackClicked}>
        <img className="NavImg" src={backBtn} alt="Back" />
      </button>
    </div>
  );
};

ActionBarComp.defaultProps = {
  showAdd: true,
  headerText: 'default',
  onAdd: () => { },
  onBack: () => { },
};

ActionBarComp.propTypes = {
  showAdd: PropTypes.bool,
  headerText: PropTypes.string,
  onAdd: PropTypes.func,
  onBack: PropTypes.func,
};
export default ActionBarComp;
