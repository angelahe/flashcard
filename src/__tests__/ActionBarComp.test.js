import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import ActionBarComp from '../components/ActionBarComp';

it('renders without crashing', () => {
  shallow(<ActionBarComp />);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ActionBarComp />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('snapshot of UI renders consistently', () => {
  const tree = renderer.create(<ActionBarComp />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('contains the container div ActionBarComp on shallow render', () => {
  const component = shallow(<ActionBarComp />);
  expect(component.html()).toContain('ActionBarComp');
  expect(component.html()).toContain('AppLogo');
  expect(component.html()).toContain('AddBtn');
  expect(component.html()).toContain('BackBtn');
  expect(component.html()).toContain('HeaderText');
});

it('does not have add button if showAdd is false', () => {
  const handleAddClick = jest.fn(() => {});
  const handleBackClick = jest.fn(() => {});
  const component = shallow(<ActionBarComp
    showAdd={false}
    headerText="Header"
    onAdd={handleAddClick}
    onBack={handleBackClick}
  />);
  expect(component.html()).not.toContain('AddBtn');
});

it('calls addHandler on add click', (done) => {
  const handleAddClick = jest.fn(() => {});
  const handleBackClick = jest.fn(() => {});
  const component = shallow(<ActionBarComp
    showAdd
    headerText="Header"
    onAdd={handleAddClick}
    onBack={handleBackClick}
  />);
  const button = component.find('.AddBtn').first();
  button.simulate('click');

  setTimeout(() => {
    expect(handleAddClick.mock.calls.length).toBe(1);
    done();
  });
});

it('calls backHandler on back click', (done) => {
  const handleAddClick = jest.fn(() => {});
  const handleBackClick = jest.fn(() => {});
  const component = shallow(<ActionBarComp
    showAdd={false}
    headerText="Header"
    onAdd={handleAddClick}
    onBack={handleBackClick}
  />);
  const button = component.find('.BackBtn').first();
  button.simulate('click');

  setTimeout(() => {
    expect(handleBackClick.mock.calls.length).toBe(1);
    done();
  });
});
