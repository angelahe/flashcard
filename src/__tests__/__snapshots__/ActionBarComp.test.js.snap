// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCardComp renders correctly 1`] = `
<div>
  <span>
    Image Search:
  </span>
  <input
    onChange={[Function]}
    value=""
  />
  <button
    onClick={[Function]}
    type="button"
  >
    <img
      alt="Add"
      className="btnImg"
      src="search_FFFFFF.png"
    />
  </button>
  <br />
  <div
    className="cardContainer"
  >
    <img
      alt="nounprojectimg"
      className="cardImg"
      src=""
    />
  </div>
  <div
    className="imagesContainer"
  />
  <p>
    Image Selected Values
  </p>
  <span>
    Url: 
    
     ID: 
    
  </span>
  <button
    onClick={[Function]}
    type="button"
  >
    OK
  </button>
</div>
`;
